File: src/main/java/com/github/bgomar/consolelogger/toolwindow/ComboBoxWithImageItem.java
package com.github.bgomar.consolelogger.toolwindow;

import javax.swing.*;

public class ComboBoxWithImageItem {
    private final String title;
    private final String iconPath;

    public ComboBoxWithImageItem(String title, String iconPath) {
        this.title = title;
        this.iconPath = iconPath;
    }

    public String displayName() {
        return title;
    }

    public String getTitle() {
        return title;
    }

    public String getIconPath() {
        return iconPath;
    }

    @Override
    public String toString() {
        return title;
    }
}

File: src/main/kotlin/com/github/bgomar/consolelogger/toolWindow/ComboBoxWithImageItem.kt
package com.github.bgomar.consolelogger.toolWindow

/**
 * [ComboBoxWithImageRenderer] item: PNG or SVG image + text.
 * @param title item's text
 * @param imagePath item's image path relative to the 'resources' folder, without leading `/`. Example: `consolelogger/image.svg`
 */
data class ComboBoxWithImageItem(val title: String, val imagePath: String) {

    fun displayName(): String {
        return title
    }
}
